src/columnstore_handler.cpp

```diff
@@ -9,6 +9,38 @@ extern "C" {
 #include "utils/syscache.h"
 }
 
+namespace {
+
+// Get precision for postgres "numerics" type, copied from pg_duckdb.
+int numeric_typmod_precision(int32 typmod) {
+    return ((typmod - VARHDRSZ) >> 16) & 0xffff;
+}
+// Get scale for postgres "numerics" type, copied from pg_duckdb.
+int numeric_typmod_scale(int32 typmod) {
+    return (((typmod - VARHDRSZ) & 0x7ff) ^ 1024) - 1024;
+}
+
+// If column is of type "numeric", check whether it's acceptable for mooncake;
+// If not, exception is thrown via `elog`.
+void ValidateColumnNumericType(Form_pg_attribute &attribute) {
```

Comment:
It seems good to also have the same lockdown for pg_duckdb's temp DuckDB table?
So it might be better to make this change to pg_duckdb instead?

Reply:
If I don't read it wrong, this is how pg_duckdb handles temp table creation:
https://github.com/duckdb/pg_duckdb/blob/bb82c937936b97824bb766893745e1456eb1aa08/src/pgduckdb_ddl.cpp#L199-L364

Considering the fact that pg_duckdb doesn't support complete table creation functionality, I think it would be nice if (have to say having both plugins at the same postgres is little messy):
- pg_duckdb locks down temp file creation on duckdb-incompatible types
- pg_mooncake locks down file creation on duckdb-incompatible types

Reply:
Yes, the creation of columnstore tables and DuckDB temporary tables follow different code paths
My suggestion was only to move the helper functions `numeric_typmod_precision`, `numeric_typmod_scale`, and `ValidateColumnNumericType` in this file to pg_duckdb
I noticed you just opened https://github.com/duckdb/pg_duckdb/pull/471, which already addresses this
We just need to wait for that diff to land and re-pull from pg_duckdb